sqlglot/dialects/redshift.py

```diff
@@ -115,6 +115,12 @@ def _parse_approximate_count(self) -> t.Optional[exp.ApproxDistinct]:
             self._retreat(index)
             return None
 
+        def _parse_column(self) -> t.Optional[exp.Expression]:
+            column = super()._parse_column()
+            if column:
+                column.set("join_mark", self._match(TokenType.JOIN_MARKER))
+            return column
```

Comment:
Let's not duplicate this logic here. Instead, you can move this to the base `Parser`'s `_parse_column` method and control it using a dialect flag.

Notice that we already have a flag in the `Generator` class (see [here](https://github.com/tobymao/sqlglot/blob/main/sqlglot/generator.py#L228-L229)), so introducing another one in the `Parser` is going to be redundant. Instead, let's move that generator constant in the `Dialect` class and have Redshift and Oracle set it to `True`.

Then, you'll access it as `self.dialect.COLUMN_JOIN_MARKS_SUPPORTED` in both the parser and the generator.